avm/res/app/managed-environment/tests/e2e/defaults/main.test.bicep

```diff
@@ -49,8 +50,22 @@ module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem'
   name: '${uniqueString(deployment().name, resourceLocation)}-test-${serviceShort}-${iteration}'
   params: {
     name: '${namePrefix}${serviceShort}001'
-    location: resourceLocation
     logAnalyticsWorkspaceResourceId: nestedDependencies.outputs.logAnalyticsWorkspaceResourceId
+    location: resourceLocation
+    workloadProfiles: [
```

Comment:
I take it, all these values are required for zone redundancy? Otherwise the `defaults` test would be the wrong place to add the parameters.

Reply:
Yes, it appears that all these settings are needed to enable zone redundancy and get a successful deployment.

Reply:
@eriqua, @ChrisSidebotham , @jtracey93 I guess the parameters should then all be marked as 'required', or at least 'conditional'?

Reply:
Defaults tests should only include Required parameters. If getting the defaults test WAF reliability compliant results in adding more parameters to the defaults test, then this in turns means all added Optional parameters must become Required. 

So yes, they should be marked Required and any corresponding default value should be removed.

Reply:
Hey @eriqua, I'd vouch for `Conditional` as they're seemingly only required if the zoneRedundancy parameter is `true`. Just because it's the default, I'd argue it doesn't mean all other values are required too. I'd almost say it's a prime example for a 'conditional' parameter 🥇 
Just thinking - if one would set the parameter to `false`, all parameters would suddenly be optional and not required anymore. So that sounds conditional too me 😄

Reply:
Fair, just wondering how to deal with the defaults test definition. 
It used to cover "only required parameters". 
Now it's more in the lines of "required parameters plus conditional parameters required to make the test WAF compliant". 
Let's agree on the guidelines as this very likely will be the case for other modules, if not for waf reliability, for other pillars that we will enforce next.

Reply:
I guess if a condition for a conditional parmaeter is flipped to `true` it can become required, that's why it's 'conditionally'.

Reply:
@AlexanderSehr yep, but, for example, in case of other conditional parameters, not related to WAF, we'll ask to remove them from defaults. And this may become confusing if not properly documented. So wondering, would it help to add something like this to the WAF conditional descriptions: "Conditional. Required if <xxx> is set to <yyy>. Required to make the resource WAF compliant. <zzz>."

Reply:
Personally, I don't share the confusion (i.e., don't think it breaks with any current statements and makes sense as is), BUT, I also don't have a strong opinion about it. So, if you think it's needed, I won't stop you 😄

Reply:
@hundredacres any take from you?

I guess it would be good to get this merged soon @eriqua, so let us three try and come to an agreement soon.

Reply:
@hundredacres @AlexanderSehr my suggestion would be to add description as suggested above for all parameters added to the defaults test to make it WAF compliant.

Reply:
@eriqua @AlexanderSehr  Updated params from optional to conditional and added the text re: WAF. Let me know if that works.

Reply:
Hey @hundredacres, almost. We now waf the `Required to make the resource WAF compliant.` in there as per @eriqua's request. However, it should be slightly altered to also reference the parameter that causes the issue to begin with. That is, change the sentence across all the just updated conditional parameters to 
> Required if `zoneRedundant` is set to true and with it to achieve WAF-compliance.
Or something like that. Otherwise users would need to guess which parameter may be causeing the required status to begin with. I hope that makes sense.

Reply:
Updated.